build-automation.js

```diff
@@ -0,0 +1,137 @@
+const util = require("util")
+
+const exec = util.promisify(require("child_process").exec);
+
+const https = require("https");
+
+const mapSeries = (arr) => {
+  const length = arr.length;
+  const results = new Array(length);
+
+  arr.reduce((chain, item, i) => {
+    return chain.then(() => item).then(val => results[i] = val);
+  }, Promise.resolve())
+  .then(() => results);
+}
+
+// a function that takes an URL as argument and makes a request to that URL
+// returning the response as a promise
+const request = (url) => {
+  return new Promise((resolve, reject) => {
+    https.get(url, async (res) => {
```

Comment:
hah, long time since I saw raw node http API 😅 Might be worth considering using node 17 and `--experimental-fetch`?